lib/application.dart

```diff
@@ -138,18 +144,25 @@ class _LenraApplicationState extends State<LenraApplication> {
           future: oauth2.getToken(),
           builder: (context, snapshot) {
             if (snapshot.connectionState == ConnectionState.done) {
-              if (snapshot.hasError) {
-                return Text('Error ${snapshot.error}');
+              if (!snapshot.hasError) {
+                return SocketManager(
+                  appName: widget.appName,
+                  endpoint: widget.socketEndpoint,
+                  token: snapshot.data as AccessTokenResponse,
+                  autoConnect: widget.autoConnect,
+                  child: widget.child,
+                );
+              } else {
+                Future<void>.delayed(const Duration()).then(
```

Comment:
I'm guessing this is because you can't use setState while building.

If that is the case you could use this instead :

```suggestion
                    WidgetsBinding.instance.addPostFrameCallback((context) {
```